pkg/resources/stream_on_table_acceptance_test.go

```diff
@@ -0,0 +1,564 @@
+					assert.Check(resource.TestCheckResourceAttrSet(resourceName, "describe_output.0.stale_after")),
+					assert.Check(resource.TestCheckResourceAttr(resourceName, "describe_output.0.owner_role_type", "ROLE")),
+				),
+			},
+			// import
+			{
+				Config:       config.FromModel(t, model),
+				ResourceName: resourceName,
+				ImportState:  true,
+				ImportStateCheck: assert.AssertThatImport(t,
+					resourceassert.ImportedStreamOnTableResource(t, resourceId).
+						HasNameString(id.Name()).
+						HasDatabaseString(id.DatabaseName()).
+						HasSchemaString(id.SchemaName()).
+						HasFullyQualifiedNameString(id.FullyQualifiedName()).
+						HasAppendOnlyString(r.BooleanFalse).
+						HasTableString(table.ID().FullyQualifiedName()),
+				),
+			},
+		},
+	})
+}
+
+func TestAcc_StreamOnTable_CopyGrants(t *testing.T) {
+	acc.TestAccPreCheck(t)
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifier()
+	resourceName := "snowflake_stream_on_table.test"
+
+	var createdOn string
+
+	table, cleanupTable := acc.TestClient().Table.CreateWithChangeTracking(t)
+	t.Cleanup(cleanupTable)
+	model := model.StreamOnTable("test", id.DatabaseName(), id.Name(), id.SchemaName(), table.ID().FullyQualifiedName())
+	resource.Test(t, resource.TestCase{
+		ProtoV6ProviderFactories: acc.TestAccProtoV6ProviderFactories,
+		TerraformVersionChecks: []tfversion.TerraformVersionCheck{
+			tfversion.RequireAbove(tfversion.Version1_5_0),
+		},
+		CheckDestroy: acc.CheckDestroy(t, resources.StreamOnTable),
+		Steps: []resource.TestStep{
+			{
+				Config: config.FromModel(t, model.WithCopyGrants(true)),
+				Check: assert.AssertThat(t, resourceassert.StreamOnTableResource(t, resourceName).
+					HasNameString(id.Name()),
+					assert.Check(resource.TestCheckResourceAttrWith(resourceName, "show_output.0.created_on", func(value string) error {
+						createdOn = value
+						return nil
+					})),
+				),
+			},
+			{
+				Config: config.FromModel(t, model.WithCopyGrants(false)),
+				Check: assert.AssertThat(t, resourceassert.StreamOnTableResource(t, resourceName).
+					HasNameString(id.Name()),
+					assert.Check(resource.TestCheckResourceAttrWith(resourceName, "show_output.0.created_on", func(value string) error {
+						if value != createdOn {
+							return fmt.Errorf("view was recreated")
+						}
+						return nil
+					})),
+				),
+			},
+			{
+				Config: config.FromModel(t, model.WithCopyGrants(true)),
+				Check: assert.AssertThat(t, resourceassert.StreamOnTableResource(t, resourceName).
+					HasNameString(id.Name()),
+					assert.Check(resource.TestCheckResourceAttrWith(resourceName, "show_output.0.created_on", func(value string) error {
+						if value != createdOn {
+							return fmt.Errorf("view was recreated")
+						}
+						return nil
+					})),
+				),
+			},
+		},
+	})
+}
+
+// There is no way to check at/before fields in show and describe. That's why in we try creating with these values, but do not assert them.
```

Comment:
There's something missing in the second sentence? 😅

Reply:
Fixed.